airflow/models.py

```diff
@@ -3511,8 +3515,11 @@ def verify_integrity(self, session=None):
         task_ids = []
         for ti in tis:
             task_ids.append(ti.task_id)
-            if not dag.get_task(ti.task_id) and self.state not in State.unfinished():
-                ti.state = State.REMOVED
+            try:
+                dag.get_task(ti.task_id)
+            except AirflowException:
```

Comment:
Wonder about this logic. Is it possible to get an AirflowException for another reason, and erroneously set the state to REMOVED?